app/scripts/modules/core/src/presentation/hooks/useLatestPromise.hook.spec.tsx

```diff
@@ -0,0 +1,161 @@
+import * as React from 'react';
+import { mount } from 'enzyme';
+import { IUseLatestPromiseResult, useLatestPromise } from './useLatestPromise.hook';
+
+describe('useLatestPromise hook', () => {
+  // Remove the the refresh function for .isEqual assertions
+  function promiseState(call: IUseLatestPromiseResult<any>) {
+    const { refresh, ...state } = call;
+    return state;
+  }
+
+  function Component(props: any) {
+    const { promiseFactory, deps, onChange } = props;
+    const useLatestPromiseResult: IUseLatestPromiseResult<any> = useLatestPromise(promiseFactory, deps);
+    const { status, result, error, requestId } = useLatestPromiseResult;
+
+    React.useEffect(() => onChange(useLatestPromiseResult), [status, result, error, requestId]);
+    return <></>;
+  }
+
+  function defer() {
+    let resolve: Function, reject: Function;
+    const promise = new Promise((_resolve, _reject) => {
+      resolve = _resolve;
+      reject = _reject;
+    });
+    return { promise, resolve, reject };
+  }
+
+  it('has status NONE if no promise has been returned', () => {
+    const spy = jasmine.createSpy('onChange');
+    mount(<Component promiseFactory={() => null as any} deps={[]} onChange={spy} />);
+    expect(spy).toHaveBeenCalledTimes(1);
+
+    const result: IUseLatestPromiseResult<any> = spy.calls.mostRecent().args[0];
+    expect(promiseState(result)).toEqual({ status: 'NONE', result: undefined, error: undefined, requestId: 0 });
+  });
+
+  it('has status PENDING if a promise has been returned but has not yet resolved', () => {
+    const spy = jasmine.createSpy('onChange');
+    const deferred = defer();
+    mount(<Component promiseFactory={() => deferred.promise} deps={[]} onChange={spy} />);
+    expect(spy).toHaveBeenCalledTimes(2);
+
+    const result: IUseLatestPromiseResult<any> = spy.calls.mostRecent().args[0];
+    expect(promiseState(result)).toEqual({ status: 'PENDING', result: undefined, error: undefined, requestId: 0 });
+  });
+
+  it('has status RESOLVED if a promise resolved', async done => {
+    const spy = jasmine.createSpy('onChange');
+    const deferred = defer();
+    const component = mount(<Component promiseFactory={() => deferred.promise} deps={[]} onChange={spy} />);
+    expect(spy).toHaveBeenCalledTimes(2);
+
+    deferred.resolve('payload');
+    await deferred.promise;
+    component.setProps({});
```

Comment:
for my own education: I would've expected the promise resolving to trigger a re-render on its own. Is this a thing about the way enzyme works? i.e. it never renders unless you explicitly tell it to?

Reply:
Yeah, I think this is a thing about enzyme and I really don't understand it.  I also assumed resolving the promise should cause a re-render, but it doesn't seem to be the case (😢 ).  I also tried `component.update()` which is more explicit, but it doesn't seem to have any effect.  I learned that `setProps({})` will actually do the thing.

Reply:
huh. Seems ok in cases like this to explicitly re-render, and it does make it easier to reason about what state things will be in when you call `expect`. At the same time it seems like it could get annoying if you're testing more complex renderings interactions. Oh well 🤷‍♂